check.go

```diff
@@ -1,32 +1,67 @@
 package filenameslinter
 
 import (
+	"errors"
 	"fmt"
+	"io/fs"
 	"path"
-    "io/fs"
-    "regexp"
+	"regexp"
+	"sort"
 
 	"github.com/csunibo/synta"
 	syntaRegexp "github.com/csunibo/synta/regexp"
 )
 
-func CheckDir(synta synta.Synta, fs fs.ReadDirFS, dirPath string) (err error) {
-    entries, err := fs.ReadDir(dirPath)
+func CustomReadDir(fsys fs.FS, name string) ([]fs.DirEntry, error) {
```

Comment:
A questa hack interna diamogli un nome che inizia con la minuscola. Io la chiamerei `readDir` semplicemente.
Metterei un commento sopra del tipo:
```go
// readDir uses the `readDir` method if the filesystem implements `fs.ReadDirFS`, otherwise opens the path and parses it using the `ReadDirFile` interface. 
```